umapi_client/functional.py

```diff
@@ -329,39 +317,32 @@ def maybe_split_groups(self, max_groups):
         new_commands = []
         # return True if we split at least once
         maybe_split = False
+        valid_step_keys = ['add', 'addRoles', 'remove']
         for command in self.commands:
+            # commands are assumed to contain a single key
             step_key, step_args = next(six.iteritems(command))
-            if step_key not in [StepKeys.add.name, StepKeys.remove.name, StepKeys.addRoles.name]:
+            if step_key not in valid_step_keys:
+                new_commands.append(command)
                 continue
-            split_commands = self._split_groups(step_key, command, max_groups)
-            if len(split_commands) > 1:
-                maybe_split = True
+            split_commands = [command]
+            while True:
+                new_command = {step_key: {}}
+                if not isinstance(command[step_key], dict):
+                    break
+                for group_type, groups in six.iteritems(command[step_key]):
+                    if len(groups) > max_groups:
+                        maybe_split = True
+                        command[step_key][group_type], new_command[step_key][group_type] = \
+                            groups[0:max_groups], groups[max_groups:]
+                if new_command[step_key]:
+                    split_commands.append(new_command)
```

Comment:
Better to put `maybe_split = True` here, I think, because this is where you actually do the split.  I know this is a nit comment, but it's pretty easy to miss it up above in the inner loop (I know I did :).